### ISSUE TITLE
Boolean value in document causes erratic comparison results

### ISSUE THREAD

**Issue by uranusjr** [sdispater/tomlkit#56]
Via sarugaku/plette#9.

```pycon
>>> import tomlkit
>>> s = """
... [foo]
... value = false
... """
>>> tomlkit.parse(s) == {'foo': {'value': False}}    # This works correctly.
True
>>> tomlkit.parse(s)['foo'] == {'value': False}      # This does not.
False
```

At a quick glance, `tomlkit.items.Bool` seems to be the problem here.

```pycon
>>> d = tomlkit.parse(s)
>>> d
{'foo': [{'value': False}]}
>>> d['foo']
{'value': <tomlkit.items.Bool object at 0x10c9ac780>}
```

**Comment by Alexis-Prel** [comment]
I just stumbled upon the exact same bug.  This really is a big problem.

@uranusjr have you encountered any issue with your fix #57 so far?

**Comment by frostming** [comment]
This was fixed
